ledger-api/grpc-definitions/com/daml/ledger/api/v1/active_contracts_service.proto

```diff
@@ -56,9 +58,59 @@ message GetActiveContractsResponse {
 
   // The workflow that created the contracts.
   // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional, In the Ledger API 3.0 this field will be removed.
   string workflow_id = 2;
 
   // The list of contracts that were introduced by the workflow with ``workflow_id`` at the offset.
   // Must be a valid LedgerString (as described in ``value.proto``).
+  // In the Ledger API 3.0 this field will be removed.
+  // This is empty for multi-domain API.
   repeated CreatedEvent active_contracts = 3;
+
+  // Non-empty for multi-domain API.
+  // For each contract there could be multiple ContractStateComponent-s in the entire snapshot. These together define
+  // the state of the contract in the snapshot.
+  repeated ContractStateComponent contract_state_components = 4;
```

Comment:
Do you expect different components for the same contract to be consecutive ?

Reply:
no, on the contrary actually: ATM for me seems like order by the offset would be natural here
Please note: also no ordering defined on the API (free hand for change)

Reply:
From an implementation perspective, since the incomplete transfers and active contracts come from two places, merging everything and ordering by offset may not be the most efficient thing.
Also, what offset would you take for the ActiveContract ?